bot/exts/moderation/incidents.py

```diff
@@ -418,10 +536,106 @@ async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent) ->
-        """Pass `message` to `add_signals` if and only if it satisfies `is_incident`."""
+        """
+        Pass `message` to `add_signals` and `extract_message_links` if it satisfies `is_incident`.
+
+        If `message` is an incident report, then run it through `extract_message_links` to get all
+        the message link embeds (embeds which contain information about that particular link).
+        These message link embeds are then sent into the channel.
+
+        Also passes the message into `add_signals` if the message is an incident.
+        """
         if is_incident(message):
             await add_signals(message)
 
+            webhook_embed_list = await self.extract_message_links(message)
+            if webhook_embed_list:
+                await self.send_message_link_embeds(webhook_embed_list, message, self.incidents_webhook)
+
+    @Cog.listener()
+    async def on_raw_message_delete(self, payload: discord.RawMessageDeleteEvent) -> None:
+        """
+        Delete message link embeds for `payload.message_id`.
+
+        Search through the cache for message, if found delete it from cache and channel.
+        """
+        await self.delete_msg_link_embed(payload.message_id)
+
+    async def extract_message_links(self, message: discord.Message) -> t.Optional[t.List[discord.Embed]]:
+        """
+        Checks if there's any message links in the text content.
+
+        Then passes the the message_link into `make_message_link_embed` to format a
+        embed for it containing information about the link.
+
+        As discord only allows a max of 10 embeds in a single webhook, just send the
+        first 10 embeds and don't care about the rest.
+
+        If no links are found for the message, it logs a trace statement.
+        """
+        message_links = DISCORD_MESSAGE_LINK_RE.findall(str(message.content))
+
+        if message_links:
+            embeds = []
+            for message_link in message_links[:10]:
+                ctx = await self.bot.get_context(message)
+                embed = await make_message_link_embed(ctx, message_link[0])
+                if embed:
+                    embeds.append(embed)
+
+            return embeds
+
+        log.trace(
+            f"No message links detected on incident message with id {message.id}."
+        )
+
+    async def send_message_link_embeds(
+            self,
+            webhook_embed_list: t.List,
+            message: discord.Message,
+            webhook: discord.Webhook,
+    ) -> t.Optional[int]:
+        """
+        Send Message Link Embeds to #incidents channel.
+
+        Uses the `webhook` passed in as a parameter to send
+        the embeds in the `webhook_embed_list` parameter.
+
+        After sending each webhook it maps the `message.id`
+        to the `webhook_msg_ids` IDs in the async redis-cache.
+        """
+        try:
+            webhook_msg = await webhook.send(
+                embeds=[embed for embed in webhook_embed_list if embed],
```

Comment:
```suggestion
                embeds=[embed for embed in webhook_embed_list if embed],
```
I don't follow why this is necessary? When is the item in the list falsely?